### PR TITLE
Smarter histogram rule configs and parsing logic

### PR DESCRIPTION
**Pull request by srhaber**

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -159,42 +164,46 @@ func parseRule(src, defaultValue string, rules *RuleList) string {
...
+	h = &histogram{
+		Name: rule.Name,
+		Labels: &labelset{
+			Names:  make([]string, 0),
+			Values: make([]string, 0),
+		},
+	}
+
+	var sortedNames []string
+	for name, _ := range rule.Labels {
```
Comment (albertrdixon) [inline parser.go]: ~I assume you want `_, name` here. its `index, value`~

duh, map
then just do this to be more space efficient:
```
sortedNames := make([]string, 0, len(rule.Labels))
```

Comment (albertrdixon) [reply parser.go]: actually maybe turn this into a function?
```
func keys(src map[string]string) []string {
  dst := make([]string, 0, len(src))
  for k, _ := range src {
    dst = append(dst, k)
  }
  sort.Strings(dst)
  return dst
}
```

then you can just do below:
```
for _, name := range keys(rule.Labels) {
```

**Review by albertrdixon — COMMENTED** [review]
Looks great, just a small comment.

File: parser.go (hunk 1 above)
Comment (srhaber) [reply parser.go]: I like that

**Pull request merged**